internals-js/src/coreSpec.ts

```diff
@@ -710,10 +710,14 @@ export const LINK_VERSIONS = new FeatureDefinitions<CoreSpecDefinition>(linkIden
 registerKnownFeature(CORE_VERSIONS);
 registerKnownFeature(LINK_VERSIONS);
 
-export function removeAllCoreFeatures(schema: Schema) {
+export function removeAllCoreFeatures(schema: Schema, exposeDirectives?: string[]) {
   // Gather a list of core features up front, since we can't fetch them during
   // removal. (Also note that core being a feature itself, this will remove core
   // itself and mark the schema as 'not core').
+  const directivesToPromote = exposeDirectives || [];
+  if (!directivesToPromote.every(d => d[0] === '@')) {
+    throw new GraphQLError('Directive names must start with "@"');
```

Comment:
I felt similarly that GraphQLError didn't feel appropriate, but seemed to be the pattern so I decided to go with it. I'll change it to a custom error.

Reply:
I didn't realize that all the custom errors in error.ts are all GraphQLErrors. Will Studio expect other types of errors to be thrown? If so, is there a error type to extend for these errors?

On the directive name vallidation, is there a function for this somewhere? I wasn't able to find it (but didn't look very long).

Reply:
>  Will Studio expect other types of errors to be thrown?

I don't know. @sachindshinde might have some insight on this. I suspect that Studio could adapt to any change here (after all, they'll have to make change to the their code for this, so they can also add more error handling) but it may be that sticking to `GraphQLError` requires less change on their side. And again, ok to facilitate things if that's the case (but if so, I think an error code will definitively help).

> On the directive name vallidation, is there a function for this somewhere?

Graphql-js has a `assertName` method for this. We use it in `internals-js/src/validate.js` in particular. That said, the spec essentially says names should match `[_A-Za-z][_0-9A-Za-z]*`, so matching that regex is another option.